File: src/api/Dtos/ProvinceDto.cs
public class ProvinceDto
{
    public int Id { get; set; }
    public string Name { get; set; } = null!;
    public string ArName { get; set; } = null!;
    public List<CityDto> Cities { get; set; } = new List<CityDto>();
    public int Population { get; set; }
    public double Area { get; set; }
    public string Description { get; set; } = null!;
    public string ImageUrl { get; set; } = null!;
    public List<AttractionDto> Attractions { get; set; } = new List<AttractionDto>();
    public static ProvinceDto FromEntity(Province province)
    {
        return new ProvinceDto
        {
            Id = province.Id,
            Name = province.Name,
            ArName = province.ArName,
            Population = province.Population,
            Area = province.Area,
            Description = province.Description,
            ImageUrl = province.ImageUrl,
            Cities = CityDto.FromEntities(province.Cities.ToList()),
            Attractions = AttractionDto.FromEntities(province.Attractions.ToList())
        };
    }
    public static List<ProvinceDto> FromEntities(List<Province> provinces)
    {
        return provinces.Select(p => FromEntity(p)).ToList();
    }
}

File: src/dash/src/Province/ProvinceDto.ts
import axiosInstance from "@/shared/utils/axiosInstance";
import imageUploadEndpoint from "@/shared/utils/imageUploadEndpoint";
import z from "zod";
export const EditProvinceSchema = z.object({
  name: z.string({ error: "Name is required" }).min(1).max(100),
  arName: z.string({ error: "Arabic Name is required" }).min(1).max(100),
  description: z.string().max(500).optional(),
  arDescription: z.string().max(500).optional(),
  image: z
    .union([z.string(), z.instanceof(File, { error: "Invalid file" })])
    .optional(),
  population: z.number().min(0).optional(),
  area: z.number().min(0).optional(),
});
export type EditProvinceDto = z.infer<typeof EditProvinceSchema>;
export interface EditProvinceRequestType {
  provinceId: number;
  data: EditProvinceDto;
}
export const EditProvinceRequest = async ({
  provinceId,
  data,
}: EditProvinceRequestType) => {
  try {
    if (data.image && data.image instanceof File) {
      const imageRes = await imageUploadEndpoint(data.image);
      data.image = imageRes.fileName;
    }
    const refinedData = {
      name: data.name,
      arName: data.arName,
      description: data.description,
      arDescription: data.arDescription,
      imageUrl: data.image,
      population: data.population,
      area: data.area,
    };

    const res = await axiosInstance.put(`/province/${provinceId}`, refinedData);
    return res.data;
  } catch (error) {
    return Promise.reject(error);
  }
};
